Sources/SwiftBAMDC/SwiftMetricsBAMConfig.swift

```diff
@@ -543,63 +541,54 @@ public class BMConfig : IBAMConfig {
                     headerCopy["X-TransactionId"] = UUID().uuidString.lowercased()
                 }*/
 
-                var kitReqType = Request.Method.post
+                var request = RestRequest(method: .post, url: urlString)
 
                 if(reqType.uppercased() == "GET") {
-                    kitReqType = Request.Method.get
+                    request = RestRequest(method: .get, url: urlString)
                 }
 
```

Comment:
It would be better not to create two RestRequests here when the reguest method is "get".

Can we do:
```swift
var method = HTTPMethod.put
if (reqType.uppercased() == "GET") {
  method = HTTPMethod.get
}
let request = RestRequest(method: method, url: urlString)

```